### ISSUE TITLE
Too many simultaneous joins shouldn't crash the group

### ISSUE THREAD

**Issue by etremel** [Derecho-Project/derecho-unified#11]
In order to prevent split-brain behavior, Derecho's view manager can't accept more new members in a single view-change than the current size of the group. Right now, however, if the leader receives a join request when it already has the maximum number of pending join requests, it throws an uncaught `derecho_exception`, which completely shuts down the system. This is far too drastic a step to take. Instead, the leader should keep the join request in a queue and propose it as a change once all the currently-proposed join requests are installed.

**Comment by sagarjha** [comment]
I thought I had a solution to this. The changes and joiner_ips arrays in derechoSST have a size of 100 + number of members. This helped me start all the nodes at the same time on Fractus and Stampede as well.
Refer to my email thread "Report on experiments status (It's not yet complete. Let's talk in the meeting today)". A relevant excerpt from that is:
"
We want to allow multiple nodes to simultaneously contact the leader and be able to join the group. Handling multiple joins was implemented by Edward few months ago, but we allowed the number of changes (num_changes' size) to the view (joins, leaves, failures) to not exceed the number of members (just an arbitrary upper limit). This stops me from having all the nodes join the group at the same time (which I want to save time in the experiment scripts), because the first view contains only 2 members and as soon as more than 2 new nodes contact the leader, it throws up. I inflated this maximum limit to 100 plus the number of members. But, that unearthed some hidden bugs. ...
"
